src/EditorFeatures/Core/CommandHandlers/AbstractGoToCommandHandler.cs

```diff
@@ -0,0 +1,158 @@
+
+        protected abstract Task FindAction(TLanguageService service, Document document, int caretPosition, IFindUsagesContext context);
+
+        public AbstractGoToCommandHandler(IEnumerable<Lazy<IStreamingFindUsagesPresenter>> streamingPresenters)
+        {
+            _streamingPresenters = streamingPresenters;
+        }
+
+        public VSCommanding.CommandState GetCommandState(TCommandArgs args)
+        {
+
+            // Because this is expensive to compute, we just always say yes as long as the language allows it.
+            var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            var findUsagesService = document?.GetLanguageService<TLanguageService>();
+            return findUsagesService != null
+                ? VSCommanding.CommandState.Available
+                : VSCommanding.CommandState.Unavailable;
+        }
+
+        public bool ExecuteCommand(TCommandArgs args, CommandExecutionContext context)
+        {
+            using (context.OperationContext.AddScope(allowCancellation: true, _scopeDescription))
+            {
+                var subjectBuffer = args.SubjectBuffer;
+                if (!subjectBuffer.TryGetWorkspace(out var workspace))
+                {
+                    return false;
+                }
+
+                var service = workspace.Services.GetLanguageServices(args.SubjectBuffer)?.GetService<TLanguageService>();
+                if (service != null)
+                {
+                    var caret = args.TextView.GetCaretPoint(args.SubjectBuffer);
+                    if (caret.HasValue)
+                    {
+                        var document = subjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(
+                            context.OperationContext).WaitAndGetResult(context.OperationContext.UserCancellationToken);
+                        if (document != null)
+                        {
+                            if (service != null)
+                            {
+                                ExecuteCommand(document, caret.Value, service, context);
+                            }
+                            return true;
+                        }
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        private void ExecuteCommand(
+           Document document, int caretPosition,
+           TLanguageService service,
+           CommandExecutionContext context)
+        {
+            var streamingPresenter = GetStreamingPresenter();
+
+            // We have all the cheap stuff, so let's do expensive stuff now
+            string messageToShow = null;
+
+            var userCancellationToken = context.OperationContext.UserCancellationToken;
+            using (Logger.LogBlock(FunctionId.CommandHandler_GoToBase, KeyValueLogMessage.Create(LogType.UserAction), userCancellationToken))
```

Comment:
This will be invoked for both goto implementation and goto base right?  I'd suggest either tracking these two operations differently or using a more generic name (maybe CommandHandler_GoTo or something)

Reply:
Thought about this last night, another option is add a property to the log request here that specifies the type of GoTo, then you can just have one event logged but you'll still be able to distinguish the type of the GoTo